inference-engine/src/readers/ir_reader/ie_ir_reader.cpp

```diff
@@ -27,20 +29,21 @@ bool IRReader::supportModel(std::istream& model) const {
     const int header_size = 128;
     std::string header(header_size, ' ');
     model.read(&header[0], header_size);
+    model.seekg(0, model.beg);
+
+    pugi::xml_document doc;
+    auto res = doc.load_string(header.c_str(), pugi::parse_default | pugi::parse_fragment);
```

Comment:
Am I right that header contains some incorrect string? Something like:
`<net version="10" name="test"> <layers><lay`
or
`<net version="10" name="test"`
or
`<net name="test" versio`

Will it works for all cases?

Reply:
BTW, may be we need to extract this function (which detects version of IR) to a separate header and write unit tests for it.

Reply:
Yes, all of these cases are handled correctly.